pkg/operator/api/api.go

```diff
@@ -220,6 +220,10 @@ func (a *apiServer) ComponentUpdate(in *operatorv1pb.ComponentUpdateRequest, srv
 	}()
 	chWrapper := initChanGracefully(updateChan)
 	updateComponentFunc := func(c *componentsapi.Component) {
+		if c.Namespace != in.Namespace {
+			return
```

Comment:
How about adding a warn here then admin can find invalid calls?

Reply:
No, because this is a valid operation. It's an internal decision of the operator to not update sidecars from a different namespace, but it's not an invalid operation.